source/math/FloatConvexHull2D.ooc

```diff
@@ -57,18 +61,77 @@ FloatConvexHull2D: class {
 		this contains(box leftBottom) && this contains(box rightBottom) && this contains(box leftTop) && this contains(box rightTop)
 	}
 	transform: func (transform: FloatTransform2D) -> This {
-		// TODO: Re-compute convex hull from new points if necessary, i.e. if transform changes order of points
 		newPoints := VectorList<FloatPoint2D> new(this count)
 		for (i in 0 .. this count)
 			newPoints add(transform * this _points[i])
 		This new(newPoints)
 	}
-	_isOnLeft: static func (leftPoint, rightPoint, queryPoint: FloatPoint2D) -> Bool {
-		(rightPoint x - leftPoint x) * (queryPoint y - leftPoint y) > (rightPoint y - leftPoint y) * (queryPoint x - leftPoint x)
+	computeHull: func {
+		if (this count > 2) {
+			// Uses the Quickhull algorithm if more than two points, average complexity O(n log n)
+			// https://en.wikipedia.org/wiki/Quickhull
+			points := this _points copy()
+			this _points clear()
+			leftMostIndex := 0
+			rightMostIndex := 0
+			for (i in 0 .. points count)
```

Comment:
`for (i in 1 .. points count)` No branch condition will be true the first iteration.